src/pages/EditAndApproveStoryPage/EditAndApproveStoryPage.tsx

```diff
@@ -9,15 +30,131 @@ interface EditAndApproveStoryPageProps {
 export const EditAndApproveStoryPage: React.FC<EditAndApproveStoryPageProps> = (
   props
 ): JSX.Element => {
+  const history = useHistory();
+  const classes = useStyles();
+  const [success, setSuccess] = useState<boolean>(false);
+  const [successMessage, setSuccessMessage] = useState<string>('');
+
   /**
    * If a Prospect object was passed to the page from one of the other app pages,
    * let's extract it from the routing.
    */
-  const location = useLocation<{ prospect?: Prospect }>();
+  const location = useLocation<EditAndApproveStoryPageProps>();
+  const prospect = location.state?.prospect;
+
+  // prepare "approve" mutation
+  const [approveProspectMutation, { loading, error }] = useMutation<
+    ApproveProspectData,
+    ApproveProspectVariables
+  >(approveProspect);
+
+  /**
+   * Collect form data, choose action and send it off to the API
+   *
+   * @param data
+   */
+  const handleSubmit = (data: EditAndApproveStoryFormData) => {
```

Comment:
Could we abstract the graphql query from this component as we discussed? 
Ideally, in this component, the `handle` method would be calling functions like `approveProspect` and `snoozeProspect` with arguments.

Let me know if the plan is to do that in a follow-up PR, that works for me as well.

Reply:
Yes, the current plan is to refactor API calls in frontend code in smaller follow-up PRs: https://github.com/Pocket/curation-tools-frontend/issues/99#issuecomment-766624513. I would definitely prefer this PR to be merged more or less as is and address the imperfections in subsequent PRs.